compute_endpoint/tests/unit/test_endpointmanager_unit.py

```diff
@@ -880,6 +920,113 @@ def test_iterates_even_if_no_commands(mocker, epmanager_as_root):
     assert em._command_queue.get.called
 
 
+@pytest.mark.parametrize("hb", (-100, -5, 0, 0.1, 4, 7, 11, None))
+def test_heartbeat_period_minimum(mocker, conf_dir, hb, mock_ep_uuid, mock_reg_info):
+    conf = Config(executors=[])
+    if hb is not None:
+        conf.heartbeat_period = hb
+    em = EndpointManager(conf_dir, mock_ep_uuid, conf, mock_reg_info)
+    exp_hb = 30.0 if hb is None else max(5.0, hb)
+    assert exp_hb == em._heartbeat_period, "Expected a reasonable minimum heartbeat"
+
+
+def test_send_heartbeat_verifies_thread(mocker, conf_dir, mock_ep_uuid, mock_reg_info):
+    conf = Config(executors=[])
```

Comment:
These are good tests.  It seems you wrote a lot of this just now, instead of just adapting single user tests.  Nice, and a lot of coverage.

Reply:
That's correct.  In particular, in line with your first observation that SU and MU heartbeat codes are quite disparate currently, and not otherwise aligned with each other.